tq/basic_upload.go

```diff
@@ -74,6 +72,25 @@ func (a *basicUploadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb Progres
 	}
 	defer f.Close()
 
+	if len(req.Header.Get("Content-Type")) == 0 {
```

Comment:
Could this be extracted into its own method? I can foresee changes like adding a configuration option to disable this, changing the default content type, and etc.

Maybe something like:

```go
func setContentTypeFor(req *http.Request, r io.Reader) error { ... }
```